### ISSUE TITLE
CMake can't find spglib on Ubuntu/Debian

### ISSUE THREAD

**Issue by oschuett** [cp2k/cp2k#3414]
When building with `-DCP2K_USE_SPGLIB=ON` on Ubuntu 2024.04, I get the following error message:
```
CMake Error at CMakeLists.txt:686 (find_package):
  Could not find a package configuration file provided by "Spglib" with any
  of the following names:

    SpglibConfig.cmake
    spglib-config.cmake

  Add the installation prefix of "Spglib" to CMAKE_PREFIX_PATH or set
  "Spglib_DIR" to a directory containing one of the above files.  If "Spglib"
  provides a separate development package or SDK, be sure it has been
  installed.
```

It turns out that the [libspglib-f08-dev](https://packages.ubuntu.com/hu/noble/amd64/libspglib-f08-dev/filelist) package ships cmake files with somewhat different names:

```
/usr/lib/x86_64-linux-gnu/cmake/Spglib/SpglibTargets_fortran-none.cmake
/usr/lib/x86_64-linux-gnu/cmake/Spglib/SpglibTargets_fortran.cmake
```

@mtaillefumier what do you think?

**Comment by mtaillefumier** [comment]
We should not do anything at all except be sure that Spglib::fortran is added to the cp2k libraries dependencies. 

https://github.com/cp2k/cp2k/blob/c5411e048c1d97964d822cc470060c619c47960d/src/CMakeLists.txt#L1560C5-L1560C49

we need to add `Spglib::fortran` to this line. I can do it if you want.

**Comment by mtaillefumier** [comment]
Basically, cmake will take care of this all by itself unless the devs decided otherwise. it might be a module to be searched with `find_package(Spglib required fortran)`

**Comment by mtaillefumier** [comment]
The error is really ubuntu specific actually. `/usr/lib/x86_64-linux-gnu/cmake` should be added to `CMAKE_PREFIX_PATH`

**Comment by oschuett** [comment]
> we need to add Spglib::fortran to this line. I can do it if you want.

Yes, it would be great if you could help out with these `find_package` issues. Then I'll focus on migrating the remaining CI scripts.

**Comment by oschuett** [comment]
> we need to add Spglib::fortran to this line. I can do it if you want.

I tried this but unfortunately it didn't work. It seems to me that the [Debian/Ubuntu package](https://packages.debian.org/sid/amd64/libspglib-f08-dev/filelist) lacks the necessary cmake file. The good news is that it does have a pkgconfig file. So, we probably have to write our own `FindSpglib.cmake` script.
